src/tools/dashboard/src/util.rs

```diff
@@ -0,0 +1,203 @@
+
+/// Parses and returns the properties in a test file.
+pub fn parse_test_header(path: &Path) -> TestProps {
+    let mut fail_step = None;
+    let mut rustc_args = Vec::new();
+    let mut rmc_args = Vec::new();
+    let it = BufReader::new(File::open(path).unwrap());
+    for line in it.lines() {
+        let line = line.unwrap();
+        let line = line.trim_start();
+        if line.is_empty() {
+            continue;
+        }
+        if !line.starts_with("//") {
+            break;
+        }
+        fail_step = try_parse_fail_step(fail_step, line);
+        rustc_args = try_parse_args(rustc_args, "compile", line);
+        rmc_args = try_parse_args(rmc_args, "rmc", line);
+    }
+    TestProps::new(path.to_path_buf(), fail_step, rustc_args, rmc_args)
+}
+
+/// Adds RMC and Litani directories to the current `PATH` environment variable.
+pub fn add_rmc_and_litani_to_path() {
+    let cwd = env::current_dir().unwrap();
+    let rmc_dir = cwd.join("scripts");
+    let mut litani_dir = cwd.clone();
+    litani_dir.extend(["src", "tools", "aws-build-accumulator"].iter());
+    env::set_var(
+        "PATH",
+        format!("{}:{}:{}", rmc_dir.display(), litani_dir.display(), env::var("PATH").unwrap()),
+    );
+}
+
+/// Does RMC catch syntax, type, and borrow errors (if any)?
+pub fn add_check_job(litani: &mut Litani, test_props: &TestProps) {
+    // TODO: add logic to ensure that the next 2 steps run only if this step
+    // succeeds (and RMC is not expected to panic in this step) (see issue
+    // #392).
+    let exit_status = if test_props.fail_step == Some(FailStep::Check) { 1 } else { 0 };
+    let mut rmc_rustc = Command::new("rmc-rustc");
+    rmc_rustc.args(&test_props.rustc_args).args(["-Z", "no-codegen"]).arg(&test_props.path);
+    // TODO: replace `build` with `check` when Litani adds support for custom
+    // stages (see issue #391).
+    litani.add_job(
+        &rmc_rustc,
+        "Is this valid Rust code?",
+        test_props.path.to_str().unwrap(),
+        "build",
+        exit_status,
+    );
+}
+
+/// Is RMC expected to codegen all the Rust features in the test?
+pub fn add_codegen_job(litani: &mut Litani, test_props: &TestProps) {
+    // TODO: again, add logic to ensure that the next step runs only if this
+    // step succeeds (and RMC is not expected to panic in this step) (see issue
+    // #392).
+    let exit_status = if test_props.fail_step == Some(FailStep::Codegen) { 1 } else { 0 };
+    let mut rmc_rustc = Command::new("rmc-rustc");
+    rmc_rustc
+        .args(&test_props.rustc_args)
+        .args(["-Z", "codegen-backend=gotoc", "--cfg=rmc", "--out-dir", "build/tmp"])
+        .arg(&test_props.path);
+    // TODO: replace `test` with `codegen` when Litani adds support for custom
+    // stages (see issue #391).
+    litani.add_job(
+        &rmc_rustc,
+        "Does RMC support all the Rust features used in it?",
+        test_props.path.to_str().unwrap(),
+        "test",
+        exit_status,
```

Comment:
Input should be `test_props.path.to_str() + "build"`, output should be `test_props.path.to_str() + "test"`